celi_framework/utils/converse_client.py

```diff
@@ -0,0 +1,157 @@
+import functools
+import json
+import logging
+import os
+import boto3
+import aioboto3
+
+#from anthropic import AsyncAnthropic, AsyncAnthropicBedrock
+#from anthropic.types import ToolUseBlock, Message
+
+from celi_framework.utils.llm_response import LLMResponse, ToolCall, FunctionCall
+
+logger = logging.getLogger(__name__)
+
+
+@functools.lru_cache()
+def get_converse_bedrock_client(model_name):
+    if model_name in {
+        'anthropic.claude-3-opus-20240229-v1:0',
+        'meta.llama3-1-8b-instruct-v1:0',
+        'meta.llama3-1-70b-instruct-v1:0',
+        'meta.llama3-1-405b-instruct-v1:0',
+        'mistral.mistral-large-2407-v1:0'
+    }:
+        session = aioboto3.Session(region_name = 'us-west-2')
+    else:
+        session = aioboto3.Session(region_name = 'us-east-1')
+    return session
+
+async def converse_bedrock_chat_completion(**kwargs):
+    cleaned_kwargs = _convert_openai_to_converse_input(**kwargs)
+    async with get_converse_bedrock_client(kwargs.get('model')).client('bedrock-runtime') as client:
+        resp = await client.converse(**cleaned_kwargs)
+    return _parse_converse_response(resp)
+
+def _convert_openai_to_converse_input(**kwargs):
+    """Adjust OpenAI options to support Converse API."""
+
+    # remove arguments not supported by Converse
+    remaining_kwargs = {
+        k: v
+        for k, v in kwargs.items()
+        if k not in ["response_format", "seed", "tools", "tool_choice"] and (k != "temperature" or v is not None)
+    }
+
+    # Create inferenceConfig
+    max_tokens = kwargs.get('max_tokens', None)
+    stop_sequences = kwargs.get('stop_sequences', None)
+    temperature = kwargs.get('temperature', None)
+    top_p = kwargs.get('top_p', None)
+    inference_config = {k: v for k, v in {
+        "maxTokens": max_tokens,
+        "stopSequences": stop_sequences,
+        "temperature": temperature,
+        "topP": top_p
+    }.items() if v is not None}
+
+    # def fix_tool(t):
```

Comment:
The current behavior is to silently ignore tools if they are present.  I think it is better to throw a ValueError or assert if "tools" is in kwargs and let the user know they aren't supported.

Otherwise people could waste time trying to use these models in places where they definitely won't work.